di/main.py

```diff
@@ -456,3 +458,146 @@ def __get__(self, inst, cls):
             return self.dependencies[self.class_obj]
         except KeyError:
             raise LookupError('Unable to find an instance for {0}'.format(self.class_obj))
+
+
+class ProxyDependencyMap(DependencyMap):
+    """
+    Derivated class that implements the proper __call__ method
+    to return an InjectorProxy .
+    """
+    def __call__(self, key):
+        """ Proxy factory method.
+            >>> dm = ProxyDependencyMap()
+            >>> my_injected_dep = dm(Spam)
+        """
+        return InjectorProxy(key, self)
+
+
+class InjectorProxy(object):
+    """
+    Alternate way of using the injector with a Proxy
+
+        >>> dm = ProxyDependencyMap()
+        >>> myfoo = dm(FOO)
+
+    This code is based on the LocalProxy implemented by the Werkzeug
+    https://github.com/pallets/werkzeug/blob/master/werkzeug/local.py#L254
+    """
+
+    def __init__(self, class_obj, dependencies):
```

Comment:
:nail_care: perhaps we should switch the order of the arguments? in my mental model `dependencies` is more important than `class_obj`